### PR TITLE
`rustc_codegen_ssa` cleanups

### PR DESCRIPTION
**Pull request by nnethercote**
Just some minor improvements I found while reading through this code.

r? @jieyouxu

### CODE REVIEW

File: compiler/rustc_codegen_ssa/src/back/write.rs [hunk 1]
```diff
@@ -1851,7 +1851,7 @@ fn spawn_work<'a, B: ExtraBackendMethods>(
     .expect("failed to spawn work thread");
 }
 
-enum SharedEmitterMessage {
+pub enum SharedEmitterMessage {
```
Comment (jieyouxu) [inline compiler/rustc_codegen_ssa/src/back/write.rs:1854]: Question: I'm probably missing something, but every other visibility change is *restricting* visibility, while this particular instance is increasing visibility. Was this intentional?

**Review by jieyouxu — APPROVED** [review]
Thanks, these look like good cleanups.

**Comment by jieyouxu** [comment]
Please feel free to r=me if the default visibility -> `pub` visibility is intentional.

File: compiler/rustc_codegen_ssa/src/back/write.rs (hunk 1 above)
Comment (nnethercote) [reply compiler/rustc_codegen_ssa/src/back/write.rs:1854]: Good catch. Not sure how that happened, but I did it with `struct Diagnostic` as well. I will revert those changes.

**Comment by nnethercote** [comment]
I have reverted the unintentional `pub` additions.

@bors r=jieyouxu rollup

**Pull request merged**